### PR TITLE
Micro draft of getSupportedTypes()

### PR DESCRIPTION
**Pull request by yell0wd0g**
@foolip can you take a look at the proposed method and let me know what you think, if it's pro forma etc?

Also, can a function be static and return a Promise<> ? :)

Relates to #59.

### CODE REVIEW

**Comment by foolip** [comment]
Static methods can return promises, here are two examples I found by grepping Blink:
https://notifications.spec.whatwg.org/#dom-notification-requestpermission
https://w3c.github.io/webrtc-pc/#sec.cert-mgmt

However, in this case it seems odd that you can synchronously ask if a type is supported, but asking for the supported types is async. What kind of implementation would actually have this restriction? In other words, can't getSupportedTypes also be sync?

**Comment by yell0wd0g** [comment]
`getSupportedTypes()` might need to check which codecs are/can be e.g. hardware accelerated, whereas `isTypeSupported()` can answer questions more easily (checking on a list, essentially). 

An example: H264 is supported in Chrome/Blink in all cases in Sw and, for certain platforms and hardware configurations, by a encode accelerator. `isTypeSupported()` knows is supported, whereas `getSupportedTypes()` might need to dig into lower levels of the implementation for to bump up this format in the list.

File: MediaRecorder.html [hunk 1]
```diff
@@ -442,6 +443,20 @@
...
+
+
+          <dt><dfn><code>getSupportedTypes</code></dfn>, static</dt>
+          <dd>
+            Provides a list of the <a href="https://tools.ietf.org/html/rfc2046">
+            MIME types</a> supported by the platform. The returned list is ordered
+            so that MIME types closer to the beginning have a larger probability
+            of being supported by encode accelerators.
+            <div>
+              <em>Return type:</em> <code>sequence&lt;DOMString&gt;</code>
```
Comment (foolip) [inline MediaRecorder.html]: It returns a promise, not a sequence directly. Wording more in the style of https://notifications.spec.whatwg.org/#dom-notification-requestpermission would be nice :)

**Comment by foolip** [comment]
Ah, the order tells you additional information, I didn't read very carefully. It makes sense, concept LGTM.

**Comment by foolip** [comment]
BTW, do you have someone to more carefully review spec changes, or am I it here?

**Comment by yell0wd0g** [comment]
Too many rebases :)
Will close this PR and open another fresh one.

**Pull request closed without merging**